### ISSUE TITLE
Upgrade to docusorus v2 for Scala 2/3 code samples

### ISSUE THREAD

**Issue by julien-truffaut** [optics-dev/Monocle#1075]
Docusaurus v2 lets us define code samples in tabs, so we could have one tab for Scala 2 and one for Scala 3.

https://v2.docusaurus.io/docs/markdown-features#customizing-tabs

**Comment by yilinwei** [comment]
I will probably try using [Laika](https://github.com/typelevel/Laika) for this.

**Comment by armanbilge** [comment]
Yes, Laika supports using tabs, see for example the Toolkit docs.

https://typelevel.org/toolkit/#example
